app/src/main/java/com/example/poste/activities/FolderViewActivity.java

```diff
@@ -135,15 +93,62 @@ public void onError(String errorMessage) {
                     @Override
                     public void onItemClick(int position, Post post) {
                         PosteApplication.setSelectedPost(post);
+                        try {
+                            Intent browserIntent = new Intent(Intent.ACTION_VIEW, Uri.parse(PosteApplication.getSelectedPost().getUrl()));
+                            startActivity(browserIntent);
+                        } catch (Exception e) {
+                            e.printStackTrace();
+                            Toast.makeText(FolderViewActivity.this, R.string.internal_error, Toast.LENGTH_LONG).show();
+                        }
                     }
 
                     @Override
-                    public void onItemLongClick(int position, Post model) {
-                            // No Long click action
+                    public void onItemLongClick(int position, Post post) {
+                        PosteApplication.setSelectedPost(post);
                     }
                 },
                 PosteApplication.getSelectedFolder().getPosts()
         );
         postRecyclerView.setAdapter(postAdapter);
     }
+
+    @Override
+    public boolean onContextItemSelected(MenuItem item) {
+        Post post = postAdapter.getLocalDataSetItem();
+        Intent intent = null;
+        switch (item.getItemId()) {
+            case R.id.ctx_menu_edit_post:
+                PosteApplication.setSelectedPost(post);
+                if (PosteApplication.getSelectedPost() != null) {
+                    intent = new Intent(FolderViewActivity.this, EditPostActivity.class);
+                    intent.putExtra("postID", post.getId());
+                    intent.putExtra("folderID", PosteApplication.getSelectedFolder().getId());
+                    startActivity(intent);
+                }
+                break;
+            case R.id.ctx_menu_delete_post:
+                PosteApplication.setSelectedPost(post);
+                if (PosteApplication.getSelectedPost() != null) {
+                    PostDeletionCallback postDeletionCallback = new PostDeletionCallback() {
+                        @Override
+                        public void onSuccess() {
+                            Toast.makeText(FolderViewActivity.this, R.string.post_deleted, Toast.LENGTH_LONG).show();
+                            Intent intent = new Intent(FolderViewActivity.this, FolderViewActivity.class);
+                            finish();
+                            startActivity(intent);
+                        }
+
+                        @Override
+                        public void onError(String errorMessage) {
+                            Toast.makeText(FolderViewActivity.this, errorMessage, Toast.LENGTH_LONG).show();
+                        }
+                    };
+                    User.getUser().deletePostFromServer(PosteApplication.getSelectedPost(), postDeletionCallback);
+                    finish();
```

Comment:
I think this finish() and recreate() should be removed; we don't want them to run before the callbacks or we will have unexpected behavior